languages/js/src/Polar.ts

```diff
@@ -142,6 +188,11 @@ export class Polar {
       results.return();
       if (done) throw new InlineQueryFailedError(source);
     }
+
+    // TODO(gj): Make sure this is correct.
```

Comment:
This wouldn't do anything because the inline queries get added when you call `enable_roles`. It's technically ok and any config stuff you added would work but it wouldn't be validated.

Reply:
Ahhh, okay... what do you think we should do if someone enables roles (running validation against all previously loaded policy) and then subsequently loads additional rules that might contain invalid roles predicates?

Reply:
Yeah all these different possible orders of doing things really sucks. If we're worried about that for now I think we should error if they load any polar after enabling roles.

Reply:
Shall we just take this out for now then?

Reply:
> 
> 
> Yeah all these different possible orders of doing things really sucks. If we're worried about that for now I think we should error if they load any polar after enabling roles.

That's not a bad idea. If you want dynamic policy loading, it's not yet supported with roles enabled. Seems like a reasonable trade-off.

Reply:
> 
> 
> > Yeah all these different possible orders of doing things really sucks. If we're worried about that for now I think we should error if they load any polar after enabling roles.
> 
> That's not a bad idea. If you want dynamic policy loading, it's not yet supported with roles enabled. Seems like a reasonable trade-off.

(take as a follow up)